### PR TITLE
AD-204831: .NET 8.0 Code Review and Clean Up

### PR DESCRIPTION
**Pull request by hnandra**

### CODE REVIEW

File: NCS.DSS.NotificationsListener/NCS.DSS.NotificationsListener.csproj [hunk 1]
```diff
@@ -1,41 +1,39 @@
...
+	</PropertyGroup>
+	<PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
+		<DebugType>portable</DebugType>
+		<DebugSymbols>true</DebugSymbols>
+	</PropertyGroup>
+	<ItemGroup>
+		<PackageReference Include="DFC.HTTP.Standard" Version="0.1.11" />
+		<PackageReference Include="Microsoft.ApplicationInsights.WorkerService" Version="2.22.0" />
+		<PackageReference Include="Microsoft.Azure.Cosmos" Version="3.46.0" />
+		<PackageReference Include="Microsoft.Azure.Functions.Extensions" Version="1.1.0" />
```
Comment (ajones-DFE) [inline NCS.DSS.NotificationsListener/NCS.DSS.NotificationsListener.csproj:17]: This needs to be removed

File: NCS.DSS.NotificationsListener/NotificationsListener/Function/NotificationsListenerHttpTrigger.cs [hunk 2]
```diff
@@ -1,86 +1,83 @@
...
+                    _logger.LogError("Notification object is NULL after deserialization.");
+                    return new UnprocessableEntityResult();
+                }
             }
             catch (JsonException ex)
             {
-                _logger.LogError(ex, "Failed to deserialize the notification.");
+                _logger.LogError(ex, "Failed to deserialize the Notification from the request body. Exception {ErrorMessage}", ex.Message);
                 return new UnprocessableEntityObjectResult(_dynamicHelper.ExcludeProperty(ex, PropertyToExclude));
             }
```
Comment (ajones-DFE) [inline NCS.DSS.NotificationsListener/NotificationsListener/Function/NotificationsListenerHttpTrigger.cs:67]: Would be good to add a success log for the getResourceFromRequest invocation above

File: NCS.DSS.NotificationsListener/NotificationsListener/Function/NotificationsListenerHttpTrigger.cs [hunk 3]
```diff
@@ -91,49 +88,68 @@ public async Task<IActionResult> Run([HttpTrigger(AuthorizationLevel.Anonymous,
...
             catch (Exception ex)
             {
-                _logger.LogError(ex, "Error saving notification to the database.");
-                return new InternalServerErrorResult();
+                _logger.LogError(ex, "An error has occurred when saving Notification to Cosmos DB. Exception {ErrorMessage}", ex.Message);
+                
+                return new JsonResult(new JsonSerializerOptions())
+                {
+                    StatusCode = (int)HttpStatusCode.InternalServerError
+                };
```
Comment (ajones-DFE) [inline NCS.DSS.NotificationsListener/NotificationsListener/Function/NotificationsListenerHttpTrigger.cs:147]: Again would be good to include a success log for saving to cosmos

File: NCS.DSS.NotificationsListener/NCS.DSS.NotificationsListener.csproj (hunk 1 above)
Comment (hnandra-DFE) [reply NCS.DSS.NotificationsListener/NCS.DSS.NotificationsListener.csproj]: Fixed in commit: https://github.com/SkillsFundingAgency/dss-notificationslistener/commit/466900da31241137699ae68ccd497e9b989af88f

**Review by ajones-DFE — APPROVED** [review]
LGTM after comments resolved

**Pull request merged**